encoder/libav_encoder.cpp

```diff
@@ -321,21 +331,69 @@ LibAvEncoder::~LibAvEncoder()
 	LOG(2, "libav: codec closed");
 }
 
+
 void LibAvEncoder::EncodeBuffer(int fd, size_t size, void *mem, StreamInfo const &info, int64_t timestamp_us)
 {
 	AVFrame *frame = av_frame_alloc();
+
 	if (!frame)
 		throw std::runtime_error("libav: could not allocate AVFrame");
 
 	if (!video_start_ts_)
+	{
 		video_start_ts_ = timestamp_us;
+		segment_start_ts = video_start_ts_;
+	}
+	// Segmentation Function
+	if ((options_->segment && (timestamp_us - segment_start_ts)/ 1000 > options_->segment) && output_ready_)
+	{
+		std::scoped_lock<std::mutex> lock(reset_mutex_); // prevent other threads from trying to encode any frames
+		// Flush the encoder, finish segment, save file
+		segment_num += 1;
+		deinitOutput();
+		// Now, start up again
+		segment_start_ts = timestamp_us;
+		initOutput();
+	}
+	// Split Function
+	//want to catch the 'rising edge' of the feed_frames variable. when it rises, we want to save the file and start with a new one
+	if (options_->split && (previous_feed_value != feed_encoder_frames) && feed_encoder_frames)
+	{	
+		std::scoped_lock<std::mutex> lock(reset_mutex_);
+		// Flush the encoder, finish segment, save file.
+		segment_num += 1;
+		deinitOutput();
+		// Now, start up again
+		virtual_video_ts = 
+				timestamp_us; // When we restart recording to new file, the timestamp must start from zero again
+		initOutput();
+	}
```

Comment:
Both ``if`` blocks above do mostly the same thing.  Can we factor the common code out into a small helper function - perhaps by passing in either ``segment_start_ts`` or ``virtual_video_ts``?